src/components/Article/ArticlesCurrentUser.jsx

```diff
@@ -0,0 +1,211 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { PropTypes } from 'prop-types';
+import moment from 'moment';
+import { onUserDeleteArticle } from '../../redux/actions';
+import Toast from '../common/Toast/Toast';
+import Button from '../common/Button/Button';
+
+export class ArticlesCurrentUser extends Component {
+  state = {
+    articleStatus: 'unpublished',
+    modalActive: false,
+    showToast: false,
+    deleteStatus: 'success',
+    deleteMessage: '',
+    article: '',
+  };
+
+  renderDate = date => `Published On: ${moment(date).format('LLLL')}`;
+
+  toggleArticleStatus = (e) => {
+    const { status } = e.target.dataset;
+    this.setState({ articleStatus: status, showToast: false });
+  };
+
+  openModal = (e) => {
+    const { index } = e.target.dataset;
+    const {
+      currentUser: { articles },
+    } = this.props;
+    this.setState({
+      article: articles[index],
+      modalActive: true,
+      showToast: false,
+    });
+  };
+
+  closeModal = () => {
+    this.setState({ modalActive: false });
+  };
+
+  onDeleteArticle = () => {
+    const { article } = this.state;
+    const { deleteArticle } = this.props;
+    deleteArticle({ articleSlug: article.slug })
+      .then((message) => {
+        this.setState({
+          modalActive: false,
+          deleteStatus: 'success',
+          deleteMessage: message,
+          showToast: true,
+        });
+      })
+      .catch((errMessage) => {
+        this.setState({
+          modalActive: false,
+          deleteStatus: 'error',
+          deleteMessage: errMessage,
+          showToast: true,
+        });
+      });
+  };
+
+  renderModal = () => {
```

Comment:
Why not create a component for Modal that can be re-used?

Reply:
@kimotho-njoki  I thought about it but we don't have any other place where we are going to use it. Should I still make it a component to only be reusable once?

Reply:
You create a reusable component in case one day someone else might want to reuse it. It's good practice to avoid refactoring later on.

Reply:
@kimotho-njoki Thanks. I am working on it right away.